bugsnag-android-core/src/main/java/com/bugsnag/android/internal/MemoryMappedOutputStream.kt

```diff
@@ -0,0 +1,61 @@
+package com.bugsnag.android.internal
+
+import java.io.File
+import java.io.OutputStream
+import java.io.RandomAccessFile
+import java.nio.MappedByteBuffer
+import java.nio.channels.FileChannel
+
+/**
+ * OutputStream wrapper around a shared memory file.
+ *
+ * Memory-mapped files are maintained by the operating system, and so they are guaranteed to match
+ * the last data written to the memory region, even if the app crashes.
+ *
+ * The memory mapped file is fixed to the length specified at instantiation, meaning that
+ * write() will throw java.nio.BufferOverflowException if the file is too full.
+ */
+class MemoryMappedOutputStream(
+    file: File,
+    private val bufferSize: Int,
+    private val clearedByteValue: Byte
```

Comment:
Could this use a default argument of 0x00, or some arbitrary value, if it doesn't matter too much what the value is?